src/features/query-bar/QueryBar.js

```diff
@@ -27,12 +27,17 @@ const QueryBar = ({
     onFilterChange,
     onTemplateChange,
     templates,
-}: Props) => (
-    <section className="metadata-view-query-bar">
-        <TemplateButton activeTemplate={activeTemplate} onTemplateChange={onTemplateChange} templates={templates} />
-        <FilterButton columns={columns} conditions={conditions} onFilterChange={onFilterChange} />
-        <ColumnButton columns={columns} onColumnChange={onColumnChange} template={activeTemplate} />
-    </section>
-);
+}: Props) => {
+    const metadataColumns = columns && columns.filter(column => column.source !== 'item');
+    const columnsWithoutItemName =
+        columns && columns.filter(column => column.property !== 'name' || column.source === 'metadata');
```

Comment:
It took me a few minutes to run through the different permutations. I'm not confident this is free of bugs.

I'd like to see the logic aligned with the variable name -> columns.filter(column => !isItemName(column)).